libs/dal/src/dal.service.ts

```diff
@@ -1,16 +1,16 @@
 import mongoose, { Connection, ConnectOptions } from 'mongoose';
 
+export const baseConfig: ConnectOptions = {
+  autoIndex: process.env.MONGO_AUTO_CREATE_INDEXES === 'true',
```

Comment:
This is a breaking change for the self-hosted users do we want to use AUTO_CREATE_INDEXES as well for couple of version and then deprecated it?

```suggestion
  autoIndex: process.env.AUTO_CREATE_INDEXES === 'true' || process.env.MONGO_AUTO_CREATE_INDEXES === 'true',
```